### PR TITLE
lxc/init.go: remove for-loop in create()

### PR DESCRIPTION
**Pull request by TimRots**
* remove unneeded for-loop in create()

Signed-off-by: Tim Rots <tim.rots@protonmail.ch>

### CODE REVIEW

File: lxc/init.go [hunk 1]
```diff
@@ -84,16 +84,15 @@ func (c *cmdInit) create(conf *config.Config, args []string) (lxd.InstanceServer
...
 	// If stdin isn't a terminal, read text from it
 	if !termios.IsTerminal(getStdinFd()) {
 		contents, err := ioutil.ReadAll(os.Stdin)
 		if err != nil {
 			return nil, "", err
 		}
-
-		err = yaml.Unmarshal(contents, &stdinData)
-		if err != nil {
+		if err = yaml.Unmarshal(contents, &stdinData); err != nil {
```
Comment (stgraber) [inline lxc/init.go:95]: The LXD coding style is to avoid this syntax and keep it on two lines instead.

File: lxc/init.go [hunk 2]
```diff
@@ -292,14 +288,12 @@ func (c *cmdInit) create(conf *config.Config, args []string) (lxd.InstanceServer
 			Quiet:  c.global.flagQuiet,
 		}
 
-		_, err = op.AddHandler(progress.UpdateOp)
```
Comment (stgraber) [inline lxc/init.go:295]: Same as above, keep it on two lines.

File: lxc/init.go [hunk 3]
```diff
@@ -292,14 +288,12 @@ func (c *cmdInit) create(conf *config.Config, args []string) (lxd.InstanceServer
...
-		if err != nil {
+		if _, err = op.AddHandler(progress.UpdateOp); err != nil {
 			progress.Done("")
 			return nil, "", err
 		}
 
-		err = utils.CancelableWait(op, &progress)
```
Comment (stgraber) [inline lxc/init.go:301]: Same here

File: lxc/init.go [hunk 4]
```diff
@@ -320,8 +314,7 @@ func (c *cmdInit) create(conf *config.Config, args []string) (lxd.InstanceServer
 			return nil, "", err
 		}
```
Comment (stgraber) [inline lxc/init.go:316]: And here

**Review by stgraber — CHANGES_REQUESTED** [review]
The for loop change looks good but the single line define+check is something we're avoiding on purpose in the LXD codebase.

**Comment by TimRots** [comment]
> The for loop change looks good but the single line define+check is something we're avoiding on purpose in the LXD codebase.

Thank you for pointing this out. 
Updated the PR to revert those changes.

**Comment by stgraber** [comment]
jenkins: test this please

**Pull request merged**